src/hooks.server.ts

```diff
@@ -81,6 +82,35 @@ export const handle: Handle = async ({ event, resolve }) => {
 		}
 	}
 
+	// simple API usage limit
+	if (
+		user &&
+		event.request.method === "POST" &&
+		event.url.pathname.startsWith(`${base}/conversation`)
+	) {
+		// check that user.lastSeenAt is not from today
+		// if it is, check that user.requestCount is not over 1000
+		if (
+			user.lastSeenAt &&
+			addDays(user.lastSeenAt, 1) > new Date() &&
+			user.requestCount > 1000 // TODO: review this number when we have stats in DB
+		) {
+			return errorResponse(429, ERROR_MESSAGES.overRateLimit);
+		} else {
+			await collections.users.updateOne(
+				{ _id: user._id },
+				{
+					$inc: {
+						requestCount: 1,
+					},
+					$set: {
+						lastSeenAt: new Date(),
+					},
+				}
+			);
```

Comment:
In case we go the other way:

- `user.lastSeenAt` should be renamed to `lastMessageWrittenAt`
- The current code requires the user to wait a whole 24 hours after last message, rather than waiting until next day or 24 hours after first message
- The `requestCount` isn't actually reset, it should be reset  to `0` when going the next day

Reply:
no strong opinion either!

Reply:
pick whatever's the simplest technically, IMO

Reply:
I think the collection solution is less error prone (even if the other one is much shorter w/ code)